src/core/libmaven/mzAligner.cpp

```diff
@@ -484,23 +485,25 @@ void Aligner::alignSampleRts(mzSample* sample, vector<float> &mzPoints,ObiWarp&
     }
 }
 
-void Aligner::alignWithObiWarp(vector<mzSample*> samples,  ObiParams* obiParams, int referenceSampleIndex) {
-    std::cerr<<"Aligning Sample Retention times..."<<std::endl;
 
-    if(referenceSampleIndex < 0){
-        /**
-         * currently reference sample is choosen randomly,
-         * TODO: give user options to choose reference sample and pass index of
-         * that sample as referenceSampleIndex
-         */
+void Aligner::setRefSample(mzSample* sample)
+{
+    if(sample != nullptr)
```

Comment:
No need to print the sample name, no? This could be an inline function, in that case.